### ISSUE TITLE
update region selector

### ISSUE THREAD

**Issue by MrBoringCom** [BeepIsla/valve-region-selector#31]
When I clicked enable, it just says starting on status. please update the region selector 
https://imgur.com/a/QLs7Q39

**Comment by raulzitoe** [comment]
Yeah, I have the same problem after Steam's latest update. Not working anymore.

**Comment by DREW1POWER** [comment]
same Steam update

**Comment by BeepIsla** [comment]
I just tested this and Steam *does* start for me, the status still says "Starting" but technically it works, do you have the same behaviour or does Steam not start at all?

**Comment by raulzitoe** [comment]
Same behavior but I cannot connect to South-American servers where my friends play anymore. I am in Canada and it keeps connecting me to Virginia (expected without the region selector)

Edit: I play Team Fortress 2 btw

**Comment by BeepIsla** [comment]
Yeah, Steam starts but its network doesn't get proxied. The way this program works is by creating a local proxy server and modifying the data your client sends to Steam about your pings. For some reason the Steam client no longer connects to that proxy, it just entirely ignores it and I have no idea why...

**Comment by raulzitoe** [comment]
The latest Steam update messed with networking sockets. Not sure how it would affect your code tho.
https://store.steampowered.com/oldnews/125571

**Comment by BeepIsla** [comment]
That is unrelated, its for developers on Steam. I am having trouble with internal code changes,, unsure how I can fix this..

**Comment by DREW1POWER** [comment]
> I just tested this and Steam _does_ start for me, the status still says "Starting" but technically it works, do you have the same behaviour or does Steam not start at all?

not work, i all time play at Stokholm servers, now isnt work, im every time connect to japan by default steam connect

**Comment by bruhmomentobama** [comment]
So I kept trying till I found this issue thread, I then exited out of steam region selector and called it a day, the next day I launched up tf2 and I could not at any point find a Virginia server for the LIFE OF ME, I live in Ecuador, so my ideal servers are Peru and Chile, I see this as a total win! I don't gotta worry about region selector anymore since I'm permanently bound to Peru and Chile.

**Comment by ezpzbreezy** [comment]
have you come up with any ideas on how to fix this? matchmaking in NA is broken so being able to select which EU server I get (150 vs 180 ping) is really helpful. Hopefully you can fix this for all players in these circumstances, we appreciate it :D

**Comment by BeepIsla** [comment]
Sadly not. Steam maybe removed the launch options I used to make this work, I don't know.

**Comment by MrBoringCom** [comment]
so this region selector is outdated or you're trying to find another way?

**Comment by BeepIsla** [comment]
I haven't tested a lot yet but currently I don't really know how to solve it.

**Comment by PeterMoubarak1** [comment]
Can you fix it. Csgo on NA is garbage

**Comment by BeepIsla** [comment]
If it would be that easy it would already be fixed.

**Comment by twilightofsun** [comment]
hey there, sorry for a bit of a bump but have you made any progress on this??

this Region Selector has been my lifeline for playing TF2 with friends around the world because Valve dont like letting people do that apparently

I would give all the money in the world (or close to) for a fix for this program

**Comment by BeepIsla** [comment]
I haven't really looked yet, currently it appears as if Valve removed some launch options I use in the program. So it stopped working. I can only really think of three possible solutions:

1. Inject into the Steam process, although who knows how VAC would react
2. Block using the firewall, although Linux would probably get left behind and you would still have 200 ping to other regions so queue times might be long
3. Middleman Steam's server-list request, which requires a root certificate (Which is a security hole) and do that way, probably also leaves Linux behind and is not something you should really do outside of development for a short period of time

So all in all... I would say there isn't really a lot I can do anymore

**Comment by twilightofsun** [comment]
thankyou for your speedy reply

if that is indeed the case then I suppose it simply is what it is, I do hope that there is a way somehow to reinitialize this lovely program again

thankyou regardless <3

**Comment by twilightofsun** [comment]
i've been attempting to play games with my friends across the seas and unfortunately they've been miserable queues with no games for hours, it's really depressing and makes me realize how much I really relied on this program

so i've come to ask, would you be willing to do some research to see if there is any way for this program to come alive again for a bounty of sorts?

**Comment by BeepIsla** [comment]
All of my heads are in https://github.com/BeepIsla/valve-region-selector/issues/31#issuecomment-1111389476 and none of them are good solutions. As long as that is the current state there won't be any progress.

**Comment by raulzitoe** [comment]
@twilightofsun just to let you know, I was able to play with my friends in Brazil via the TF2 matchmaking by paying a VPN, since Valve messed up with BeepIsla program. I am sure you can use a free one but you will get a lot of lag. I am paying ExitLag and it actually improved my ping with the servers as you can choose the route and it also does not re-route your whole internet, just the games you select. Didn't want it to sound like an ad but if you really wanna play with your friends that might be the way to go.

**Comment by omperus** [comment]
> I haven't really looked yet, currently it appears as if Valve removed some launch options I use in the program. So it stopped working. I can only really think of three possible solutions:
> 
> 1. Inject into the Steam process, although who knows how VAC would react
> 2. Block using the firewall, although Linux would probably get left behind and you would still have 200 ping to other regions so queue times might be long
> 3. Middleman Steam's server-list request, which requires a root certificate (Which is a security hole) and do that way, probably also leaves Linux behind and is not something you should really do outside of development for a short period of time
> 
> So all in all... I would say there isn't really a lot I can do anymore

https://github.com/Jyben/csgo-mm-server-picker

This program for CSGO seems to still be working, maybe you can get some inspiration from there?
Thanks for your work, TF2 is basically unplayable without your efforts, would love for an update even if it only works on windows for now.

**Comment by BeepIsla** [comment]
It just blocks all IP addresses in your Firewall. But you wouldn't be able to play from Europe in Australia for example, because the matchmaker doesn't want to put you in a game with that high ping, even if you have your max ping setting set to 350.

It *works* but only for some scenarios and its not optimal.

**Comment by MrBoringCom** [comment]
> > I haven't really looked yet, currently it appears as if Valve removed some launch options I use in the program. So it stopped working. I can only really think of three possible solutions:
> 
> > 
> 
> > 1. Inject into the Steam process, although who knows how VAC would react
> 
> > 2. Block using the firewall, although Linux would probably get left behind and you would still have 200 ping to other regions so queue times might be long
> 
> > 3. Middleman Steam's server-list request, which requires a root certificate (Which is a security hole) and do that way, probably also leaves Linux behind and is not something you should really do outside of development for a short period of time
> 
> > 
> 
> > So all in all... I would say there isn't really a lot I can do anymore
> 
> 
> 
> https://github.com/Jyben/csgo-mm-server-picker
> 
> 
> 
> This program for CSGO seems to still be working, maybe you can get some inspiration from there?
> 
> Thanks for your work, TF2 is basically unplayable without your efforts, would love for an update even if it only works on windows for now.

it used to work for my back then, but it stopped working for me. idk why

**Comment by omperus** [comment]
> It just blocks all IP addresses in your Firewall. But you wouldn't be able to play from Europe in Australia for example, because the matchmaker doesn't want to put you in a game with that high ping, even if you have your max ping setting set to 350.
> 
> It _works_ but only for some scenarios and its not optimal.

I think that would be great, at least for my usecase.
I'm from SA, my main problem with matchmaking is that whenever I'm in a party with someone from "NA" (Mexico, El Salvador, Puerto Rico, etc) the matchmaking **ALWAYS** puts me in Virginia (US), even if only 1 person out of 6 from the party is from NA, we are **ALL** forced to play with 150+ ping even though a server like Lima (Peru) would be the reasonable middle ground.
Obviously the ideal solution would be for Valve to actually care about their game and make the matchmaking ping limit (I have it set to 130) work as it was intended and respect the party leader matchmaking settings, but alas.
The only solution I had for this was your program which worked perfectly if I only allowed Lima, Sao Paulo and Santiago.
Maybe you can explain how TF2 NA servers can be blocked manually on the firewall if you think updating the program isn't worth it? Would really appreciate it, thanks for all your work, sincerely appreciate it 💖

EDIT: Just thought about bots and how they can enter any server on the world they want, but maybe the method they use is banneable?

**Comment by ghost** [comment]
> > It just blocks all IP addresses in your Firewall. But you wouldn't be able to play from Europe in Australia for example, because the matchmaker doesn't want to put you in a game with that high ping, even if you have your max ping setting set to 350.
> > It _works_ but only for some scenarios and its not optimal.
> 
> I think that would be great, at least for my usecase. I'm from SA, my main problem with matchmaking is that whenever I'm in a party with someone from "NA" (Mexico, El Salvador, Puerto Rico, etc) the matchmaking **ALWAYS** puts me in Virginia (US), even if only 1 person out of 6 from the party is from NA, we are **ALL** forced to play with 150+ ping even though a server like Lima (Peru) would be the reasonable middle ground. Obviously the ideal solution would be for Valve to actually care about their game and make the matchmaking ping limit (I have it set to 130) work as it was intended and respect the party leader matchmaking settings, but alas. The only solution I had for this was your program which worked perfectly if I only allowed Lima, Sao Paulo and Santiago. Maybe you can explain how TF2 NA servers can be blocked manually on the firewall if you think updating the program isn't worth it? Would really appreciate it, thanks for all your work, sincerely appreciate it 💖
> 
> EDIT: Just thought about bots and how they can enter any server on the world they want, but maybe the method they use is banneable?

Perhaps the method was banned.
Even Steam Router Tool (IP address blocker) no longer works for TF2. The queue goes on for hours and game isnt found.

**Comment by MrBoringCom** [comment]
will there be a new application similar to this?

**Comment by RiderUA** [comment]
@BeepIsla  Hi, Isla. I find your programm [valve-region-selector], its look cool and i like it. But looks like a few days ago its stop working. When program start, the list of servers is empty, just a white sheet. Is it possible to fix somehow?
Screenshot
![image](https://user-images.githubusercontent.com/117541042/200138685-73b19ffd-a364-4010-bf6d-796b1dd724a8.png)

https://prnt.sc/xS8Oz344ivQY

**Comment by BeepIsla** [comment]
It has not worked for over half a year

**Comment by RiderUA** [comment]
> 
Oh? It's a pity
A week ago at least there was a list of servers :)
There are no other ways to select servers?

**Comment by BeepIsla** [comment]
You had a list of servers but the sliders didn't do anything, now it just broke *more*
